src/nemos/basis.py

```diff
@@ -659,7 +692,9 @@ def _compute_features(self, *xi: ArrayLike) -> FeatureMatrix:
         Raises
         ------
         ValueError:
-            If an invalid mode is specified or necessary parameters for the chosen mode are missing.
+            - If an invalid mode is specified or necessary parameters for the chosen mode are missing.
+            - In mode "conv", if the number of inputs to be convolved, doesn't match the number of inputs
```

Comment:
The init of `mode="conv"` and `mode="eval"` starts to diverge  even more, and this is another argument in favor of having distinct classes for basis that performs convolution and basis for evaluation. 

Separate PR I would say, since there is a lot to change if we include the test. Do you have suggestion on an API that may work and also naming conventions?

Most of the machinery is shared, so probably a base class which they both inherit from, with an abstract method `compute_features`, and the two variants simply re-implement that

Reply:
Refer to #202 for a discussion on the basis API